stats/lib/platform-stats-fetchers.js

```diff
@@ -53,3 +53,27 @@ export const fetchDailyRewardTransfers = async (pgPool, filter) => {
   `, [filter.from, filter.to])
   return rows
 }
+
+/**
+ * @param {Queryable} pgPool
+ * @param {import('./typings.js').DateRangeFilter} filter
+ */
+export const fetchTopEarningParticipants = async (pgPool, filter) => {
+  const { rows } = await pgPool.query(`
+    WITH latest_scheduled_rewards AS (
+      SELECT DISTINCT ON (participant_address) participant_address, scheduled_rewards
+      FROM daily_scheduled_rewards
+      ORDER BY participant_address, day DESC
+    )
+    SELECT 
+      COALESCE(drt.to_address, lsr.participant_address) as participant_address, 
+      COALESCE(SUM(drt.amount), 0) + COALESCE(lsr.scheduled_rewards, 0) as total_rewards
+    FROM daily_reward_transfers drt
+    FULL OUTER JOIN latest_scheduled_rewards lsr
+      ON drt.to_address = lsr.participant_address
+    WHERE drt.day >= $1 AND drt.day <= $2 OR drt.day IS NULL
+    GROUP BY COALESCE(drt.to_address, lsr.participant_address), lsr.scheduled_rewards
+    ORDER BY total_rewards DESC
```

Comment:
should this query have a limit? Right now it's outputting all participants ordered by earnings (11,118 rows), not just the top ones

Reply:
@PatrickWoodhead and @bajtos mentioned that we wanted users to have the ability to search their address, so I think we'd need the full list output, same as in the other leaderboard PR.

Correct me if I'm wrong!

Reply:
Yes, we want users to be able to search using their address to see their rank.